### PR TITLE
Simplify timetable

### PR DESCRIPTION
**Pull request by li-kai**
Well technically this PR brings in CSS Modules as well.

This refactor aims to reduce the amount of code needed to display the timetable, as well as to fix existing bugs such as offset in safari.

Removing the need of a background seems to be doing great for now, although I have seen some odd cases where the percentage is off by 1-3px. Thoughts?

### CODE REVIEW

File: v3/src/js/views/timetable/TimetableContainer.jsx [hunk 1]
```diff
@@ -144,68 +148,43 @@ class TimetableContainer extends Component<Props> {
...
                 lessons={arrangedLessonsWithModifiableFlag}
-                horizontalOrientation={isHorizontalOrientation}
+                isVerticalOrientation={isVerticalOrientation}
                 onModifyCell={this.modifyCell}
-                ref={(r) => { this.timetableDom = r && r.timetableDom; }}
+                ref={(r) => {
+                  this.timetableDom = r && r.timetableDom;
+                }}
               />
               <br />
```
Comment (ZhangYiJiang) [inline v3/src/js/views/timetable/TimetableContainer.jsx]: Can probably get rid of this

File: v3/src/js/views/timetable/TimetableTimings.jsx [hunk 2]
```diff
@@ -3,19 +3,24 @@ import React from 'react';
...
-    <div className="timetable-timings">
-      {_.range(props.startingIndex, props.endingIndex).map((i) => {
-        return (
-          <span key={i} className="timetable-cell">{i % 2 === 0 && convertIndexToTime(i)}</span>
-        );
+    <div className={styles.timings}>
+      {range.map((i) => {
+        const time = convertIndexToTime(i);
+        return i % 2 === 0 && i !== props.endingIndex
+          ? <time key={time} className={styles.time}>
```
Comment (ZhangYiJiang) [inline v3/src/js/views/timetable/TimetableTimings.jsx]: The indentation on this makes it look weird. Might be better to collapse down to one line or use if/else

File: v3/src/js/views/timetable/TimetableTimings.scss [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  flex-direction: column;
+  margin: $timetable-day-vertical-size 0 0 0;
+}
+
+.time {
+  transform: translateX(-50%);
+}
+
+:global(.verticalMode) .time {
+  transform: translateY(-50%);
```
Comment (ZhangYiJiang) [inline v3/src/js/views/timetable/TimetableTimings.scss]: Needs some padding-right to giving it some space from the timetable

**Comment by li-kai** [comment]
@yangshun Ping for review

File: v3/src/js/utils/timetables.js [hunk 4]
```diff
@@ -131,23 +131,25 @@ export function arrangeLessonsWithinDay(lessons: Array<Lesson>): TimetableDayArr
   if (_.isEmpty(lessons)) {
     return rows;
   }
+  lessons
+    .sort((a, b) => a.StartTime.localeCompare(b.StartTime))
+    .forEach((lesson: Lesson) => {
+      for (let i = 0, length = rows.length; i < length; i += 1) {
```
Comment (yangshun) [inline v3/src/js/utils/timetables.js]: `i++`

File: v3/src/js/views/timetable/TimetableDay.jsx [hunk 5]
```diff
@@ -3,60 +3,45 @@ import React from 'react';
...
-  const style: Object = {};
-  if (!props.horizontalOrientation) {
-    style.flexGrow = numOfRows;
-    style.WebkitBoxFlex = numOfRows;
-    style.msFlexPositive = numOfRows;
-  }
-
+  const columns = props.endingIndex - props.startingIndex;
+  const rowStyle = {
+    backgroundSize: `${400 / columns}%`,
```
Comment (yangshun) [inline v3/src/js/views/timetable/TimetableDay.jsx]: Where did the 400 come from?

File: v3/webpack/webpack.config.prod.js [hunk 6]
```diff
@@ -24,13 +35,41 @@ const productionConfig = merge([
       // will work without but this is useful to set.
       chunkFilename: '[chunkhash].js',
     },
+    module: {
+      rules: [
+        {
+          test: /\.(css|scss)$/,
+          include: parts.PATHS.styles,
+
```
Comment (yangshun) [inline v3/webpack/webpack.config.prod.js]: What's the empty line for?

File: v3/src/js/utils/timetables.js (hunk 4 above)
Comment (ZhangYiJiang) [reply v3/src/js/utils/timetables.js]: For `i++` I think eslint will complain. We can turn it off using `allowForLoopAfterthoughts: true` though - https://eslint.org/docs/rules/no-plusplus

File: v3/src/js/views/timetable/TimetableDay.jsx (hunk 5 above)
Comment (li-kai) [reply v3/src/js/views/timetable/TimetableDay.jsx]: It's `100 / (columns / 4)`. On second thought, this is a micro optimisation. I'll change it.

**Pull request merged**